source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -267,6 +273,12 @@ void AuthenticatorImpl::doneWithStatus(const Status& status) {
   ENVOY_LOG(debug, "{}: JWT token verification completed with: {}", name(),
             ::google::jwt_verify::getStatusString(status));
 
+  if (status == Status::Ok &&
+      !jwks_data_->getTokenCache()->lookupTokenCache(curr_token_->token())) {
```

Comment:
Can we not to lookup the cache twice?

Reply:
if jwt_ will not found in the cache then it should be added.